### ISSUE TITLE
tslint doesn't work well with ts 2.9 resolveJsonModule

### ISSUE THREAD

**Issue by ofrobots** [palantir/tslint#3957]
### Bug Report

TypeScript 2.9 has [support for type checking json files as modules](https://blogs.msdn.microsoft.com/typescript/2018/05/31/announcing-typescript-2-9/#json-imports). This doesn't seem to work well with `tslint` when used with a tsconfig project option. tslint attempts to apply TS lint rules to json files since they are now a part of the 'project'.

- __TSLint version__: 5.1.0
- __TypeScript version__: 2.9.1
- __Running TSLint via__: CLI and Node.js API (https://github.com/google/ts-style/issues/172)

#### TypeScript code being linted

src/index.ts
```ts
import settings from './test.json';
```

src/test.json
```json
{
  "dry": false,
  "debug": false
}
```

with `tslint.json` configuration: none provided.

with tsconfig.json:
```json
{
  "include": ["src"],
  "compilerOptions": {
    "resolveJsonModule": true
  }
}
```

#### Actual behavior

```shell
$ tslint -p tsconfig.json
The 'no-reference' rule threw an error in '/Users/ofrobots/tmp/foo/gts-test/src/test.json':
TypeError: Cannot read property 'length' of undefined
    at walk 
   ...
   ...

ERROR: /Users/ofrobots/tmp/foo/gts-test/src/test.json[1, 1]: unused expression, expected an assignment or function call
ERROR: /Users/ofrobots/tmp/foo/gts-test/src/test.json[4, 2]: Missing semicolon
```

#### Expected behavior

tslint should ignore json files.

**Comment by hbroer** [comment]
Same problem here with typescript 3.0.1. It compiles with the same errors, with and without the resolveJsonModule flag and set to true or false:

2.9.2 -> resolveJsonModule: true -> errors
2.9.2 -> resolveJsonModule: false -> ok
2.9.2 -> ~~resolveJsonModule~~ -> ok

3.0.1 -> resolveJsonModule: true -> errors
3.0.1 -> resolveJsonModule: false -> errors
3.0.1 -> ~~resolveJsonModule~~ -> errors

Not shure if this is a typescript bug with 3.0.1 ignoring the compiler flag

**Comment by iarroyo5** [comment]
Any update about this? having the same problem with TS 3.0.1 and `resolveJsonModule: true`

**Comment by mlg87** [comment]
@iarroyo5 how ive gotten around this is by putting 
```ts
declare module '*.json' {
    const value: any;
    export default value;
}
```
in `custom.d.ts`

**Comment by hbroer** [comment]
@iarroyo5 that workaround fixed the problem for me

**Comment by GabrielDelepine** [comment]
Please add in your `tslint.json` file

``` 
{
  "linterOptions": {
    "exclude": [
      "*.json",
      "**/*.json"
    ]
  }
}
```

`**/*.json` is necessary to be recursive

Available since version `5.8`

**Comment by JoshuaKGoldberg** [comment]
Seems like the `linterOptions` exclude should probably be enabled by default?

**Comment by Vaelor** [comment]
This does not work in my case:
```
ERROR: <snip>/package.json[24, 3]: " should be '
...
```
